vumi/middleware/base.py

```diff
@@ -214,14 +225,37 @@ def create_middlewares_from_config(worker, config):
     for item in config.get("middleware", []):
         keys = item.keys()
         if len(keys) != 1:
-            raise ConfigError("Middleware items contain only a single"
-                              " key-value pair. The key should be a name"
-                              " for the middleware. The value should be"
-                              " the full dotted name of the class"
-                              " implementing the middleware.")
+            raise ConfigError(
+                "Middleware items contain only a single key-value pair. The"
+                " key should be a name for the middleware. The value should be"
+                " the full dotted name of the class implementing the"
+                " middleware, or a mapping containing the keys 'class' with a"
+                " value of the full dotted class name, 'consume_priority' with"
+                " the priority level for consuming, and 'publish_priority'"
+                " with the priority level for publishing, both integers.")
         middleware_name = keys[0]
-        cls_name = item[middleware_name]
         middleware_config = config.get(middleware_name, {})
+        if type(item[middleware_name]) is str:
```

Comment:
Rather use `isinstance()` (and `basestring` instead of `str`) than directly comparing types.